### PR TITLE
Staff pick updates

### LINKED ISSUE #6724: Staff Pick label not showing up on unfiltered homepage
#### Current Behavior
- Campaign tiles on the homepage don't have the "Staff Pick" banner on them until they're filtered by the Action Finder.
#### Desired Behavior
- Staff Picks should have the banner on them at all times.
#### Steps to Replicate
- Replicated on Chrome and Safari, just loading the homepage.
#### Why This Matters

We want to make sponsors happy!
#### Relevant Screenshots + Links

screen shot 2016-07-18 at 11 03 14 am
screen shot 2016-07-18 at 11 03 23 am

### LINKED ISSUE #6789: Add Display Label Checkbox
#### User Story

As an admin, I want to toggle which campaigns get a 'staff pick' label to be displayed on the homepage. 
#### Additional Information (optional)

This needs to be translated. 
This ticket needs to be addressed for the labels to show - https://github.com/DoSomething/phoenix/issues/6724
#### Why This Matters

This will allow the staff pick label to truly reflect which campaigns are staff picked.

### PR DESCRIPTION
**Pull request by itsjoekent**
#### What's this PR do?
1. Translates the staff pick field, again. Seems like features had a derp last time?
2. Adds a new "Display staff pick" field which is under the staff pick checkbox & is translatable
3. Changes homepage logic to reference this new field when deciding to display "Staff pick"
4. Switches ^ logic to use `ds_helpers_isset` to get rid of an annoying notice.
#### How should this be reviewed?

Are staff picks properly translated now?
Does a campaign show as "STAFF PICK" on the homepage if its checked as staff pick? What happens when you check "Display staff pick label"?
#### Any background context you want to provide?

We needed to mark campaigns as staff pick without overcrowding the homepage. This was the most ideal & flexible solution we could think of.
#### Relevant tickets

Fixes ##6631
Fixes #6789 
Fixes #6724 
_did I miss any haha_
#### Checklist
- [x] Run `drush ds-global-field-fix field_staff_pick` on staging
- [x] Tested on staging.
- [x] Run `drush ds-global-field-fix field_staff_pick` on thor
- [x] Run `drush ds-global-field-fix field_staff_pick` on prod

### CODE REVIEW

**Comment by itsjoekent** [comment]
going to play around with it a bit locally just to double check everything on the #global front, @angaither can you give it a look in the meantime?

File: lib/modules/dosomething/dosomething_home/dosomething_home.module [hunk 1]
```diff
@@ -38,7 +38,7 @@ function dosomething_home_preprocess_node(&$vars) {
       // Grab all tiles attributes.
       $tiles = array();
       $tiles['nid'] = $nid;
-      $tiles['is_staff_pick'] = $node->field_staff_pick[$lang_code][0]['value'];
+      $tiles['is_staff_pick'] = dosomething_helpers_isset($node->field_staff_pick_label, $lang_code);
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_home/dosomething_home.module]: should this be checking for both is staff pick, and the label display?

**Comment by itsjoekent** [comment]
yeah found a problem with it, ignore the homepage logic for a second

**Comment by itsjoekent** [comment]
@angaither hows this? seems to work ok for US & BR

File: lib/modules/dosomething/dosomething_home/dosomething_home.module [hunk 2]
```diff
@@ -38,7 +38,13 @@ function dosomething_home_preprocess_node(&$vars) {
       // Grab all tiles attributes.
       $tiles = array();
       $tiles['nid'] = $nid;
-      $tiles['is_staff_pick'] = $node->field_staff_pick[$lang_code][0]['value'];
+
+      $display_staff_pick = FALSE;
+      if (data_get($node->field_staff_pick, $lang_code) && data_get($node->field_staff_pick_label, $lang_code)) {
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_home/dosomething_home.module:43]: isn't `data_get` a laravel function?

Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_home/dosomething_home.module:43]: we have that in :drupal: now thanks to @DFurnes or @weerd

Comment (weerd) [inline lib/modules/dosomething/dosomething_home/dosomething_home.module:43]: We don't have to live in the dark ages any longer: https://github.com/DoSomething/phoenix/pull/6702

Comment (angaither) [inline lib/modules/dosomething/dosomething_home/dosomething_home.module:43]: 💡

**Comment by angaither** [comment]
🚢

**Pull request merged**